tests/kernel/pipe/pipe/src/test_pipe.c

```diff
@@ -874,3 +880,81 @@ void test_pipe_put_min_xfer(void)
 	zassert_equal(res, 0, "did not write min_xfer");
 	zassert_true(bytes_written >= 1, "did not write min_xfer");
 }
+
+/**
+ * @brief Defining and initializing pipes at compile time
+ * @details Check if the kernel provided a mechanism for defining and
+ * initializing pipes at compile time(K_PIPE_DEFINE()).
+ * - Using the pipe which defined and initialized at compile time to
+ * transfer data and verify it.
+ * @ingroup kernel_pipe_tests
+ * @see K_PIPE_DEFINE()
+ */
+void test_pipe_define_at_compileTime(void)
+{
+	size_t written, read;
+
+	/* initialize the tx buffer */
+	for (int i = 0; i < sizeof(tx_buffer); i++) {
+		tx_buffer[i] = i;
+	}
+
+	/* get test_pipe data if something there */
+	k_pipe_get(&test_pipe, &rx_buffer, PIPE_SIZE, &read,
+			NO_CONSTRAINT, K_NO_WAIT);
```

Comment:
This doesn't seem like much of a test.  It's ignoring the return value and dropping the "read" output on the floor without checking.  I'm guessing you want to check the output for -EAGAIN at least?

Reply:
I have removed this case due to the case covered by others.